### PR TITLE
Manual

### PR DESCRIPTION
**Pull request by cdwensley**
This is more of a PS than a PR: a suggestion rather than a request, and you may only want to keep the fixed typos in the first commit.  Max suggested that AutoDoc should be used with the GAPDoc version of the KBMag manual and, as a result, I started using AutoDoc with other packages, and this PR results from my experience in doing so.  For someone wanting to convert an existing manual, the AutoDoc PackageInfo.g and makedoc.g should be at least as useful as the manual, so AutoDoc should have an Abstract; Acknowledgements and Bibliography whose layout can then be copied (the content of these is unimportant).  A new subsection 1.3.3 has been added - a checklist of things to do when making a conversion.  I have found this list very useful and AutoDoc is now used in the development versions of Utils, XMod, groupoids, IdRel and Kan.

### CODE REVIEW

File: PackageInfo.g [hunk 1]
```diff
@@ -110,13 +111,26 @@ Keywords := [ "Automatic documentation, GAP, GAPDoc" ],
...
+            "of the GNU Public License Version 2.\n"
+            ), 
+        Abstract := Concatenation( 
+            "&AutoDoc; is a &GAP; package which is meant to aid ", 
+            "&GAP; package authors in creating and maintaining the ", 
+            "documentation of their packages.\n", 
+            "<P/>\n", 
+            "(If an abstract is considered necessary, it should be included ", 
+            "in <File>PackageInfo.g</File> as a string in the ", 
+            "<C>TitlePage</C> record.\n"
```
Comment (fingolfin) [inline PackageInfo.g]: Why is the second sentence here?

File: PackageInfo.g [hunk 2]
```diff
@@ -110,13 +111,26 @@ Keywords := [ "Automatic documentation, GAP, GAPDoc" ],
 AutoDoc := rec(
     TitlePage := rec(
         Copyright := Concatenation(
-                    "&copyright; 2012-2014 by Sebastian Gutsche and Max Horn<P/>\n\n",
-                    "This package may be distributed under the terms and conditions of the\n",
-                    "GNU Public License Version 2.\n"
-                ),
+            "&copyright; 2012-2014 by Sebastian Gutsche and Max Horn<P/>\n\n",
```
Comment (fingolfin) [inline PackageInfo.g]: Should be 2017

File: doc/Tutorials.xml [hunk 3]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
 
 </Subsection>
 
+<Subsection Label="Tut:Checklist">
+<Heading>Checklist for converting an existing manual to use &AutoDoc;</Heading>
+
+Here is a checklist for authors of a package &PackageName;, 
+which already has a manual, who wish to use &AutoDoc;
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Perhaps clarify that this for packages which aleady have a GAPDoc based manual?

File: doc/Tutorials.xml [hunk 4]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
 
+<Subsection Label="Tut:Checklist">
+<Heading>Checklist for converting an existing manual to use &AutoDoc;</Heading>
+
+Here is a checklist for authors of a package &PackageName;, 
+which already has a manual, who wish to use &AutoDoc; 
+to build the manual from now on. 
+We assume that the manual is currently built by reading a file 
+<File>makedoc.g</File> and that the main <File>.xml</File> file 
+is named <File> manual.xml</File>.
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Remove extra space before ˋmanual.xmlˋ

File: doc/Tutorials.xml [hunk 5]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+<P/> 
+<List>
+<Item> 
+Copy <File>AutoDoc/makedoc.g</File> to <File>PackageName/makedoc.g</File>. 
+</Item>
+<Item>
+Edit <File>PackageName/makedoc.g</File> as follows. 
+  <List>
+  <Item> 
+  Change the header to anything suitable.
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Perhaps "header comment"?

File: doc/Tutorials.xml [hunk 6]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+  in <File>manual.xml</File>. 
+  </Item> 
+  <Item> 
+  In the <Code>gapdoc_latex_options</Code> record enter any 
+  &LaTeX; <Code>newcommands</Code> previously in <File>manual.xml</File>. 
+  (If there are none you may safely delete this record.) 
+  </Item>
+  <Item> 
+  In the <Code>entities</Code> record enter any entities 
+  previously in <File>manual.xml</File>.
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Perhaps add something like "Please refer to REF for details how to encode the entities." or so, as this is not fully obvious?

File: doc/Tutorials.xml [hunk 7]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+  </Item>
+  <Item> 
+  Replace the <Code>Copyright</Code>, <Code>Abstract</Code> and 
+  <Code>Acknowledgements</Code> fields of the <Code>TitlePage</Code> 
+  record with the corresponding material from your <File>manual.xml</File>. 
+  </Item>
+  </List> 
+</Item>
+<Item> 
+There should be a <Code>.bib</Code> file in your <Code>doc/</Code> folder
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Why? Not all packages have one. Perhaps: "If there is a .bib file ..."?

File: doc/Tutorials.xml [hunk 8]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+  Replace the <Code>Copyright</Code>, <Code>Abstract</Code> and 
+  <Code>Acknowledgements</Code> fields of the <Code>TitlePage</Code> 
+  record with the corresponding material from your <File>manual.xml</File>. 
+  </Item>
+  </List> 
+</Item>
+<Item> 
+There should be a <Code>.bib</Code> file in your <Code>doc/</Code> folder  
+(possibly <Code>manual.bib</Code> or <Code>bib.xml.bib</Code>) 
+containing the information for your list of references.
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Note that some packages also directly use XML bibliographies.

File: makedoc.g [hunk 9]
```diff
@@ -15,7 +15,16 @@ LoadPackage("AutoDoc");
...
     scaffold := rec(
-        includes := [ "Tutorials.xml", "Comments.xml" ]
+        includes := [ "Tutorials.xml", 
+                      "Comments.xml" ],
+        gapdoc_latex_options := rec( EarlyExtraPreamble := """
+            \usepackage[all]{xy} 
+            \newcommand{\bbZ} {\mathbb{Z}}
+        """ ),  
+        entities := rec( 
+            orb := "<Package>orb</Package>",
```
Comment (fingolfin) [inline makedoc.g]: Is this orb entity used anywhere?

File: PackageInfo.g (hunk 2 above)
Comment (sebasguts) [reply PackageInfo.g]: Also, should we probably use the `"""` syntax here? Would be a good place to advertise it.

File: doc/Tutorials.xml [hunk 10]
```diff
@@ -313,6 +313,97 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+<Item>
+Edit <File>PackageName/makedoc.g</File> as follows. 
+  <List>
+  <Item> 
+  Change the header to anything suitable. 
+  </Item> 
+  <Item> 
+  Replace <Code>LoadPackage("AutoDoc");</Code> 
+  with <Code>LoadPackage("PackageName");</Code>. 
+  </Item>
```
Comment (sebasguts) [inline doc/Tutorials.xml]: I think `LoadPackage("PackageName");` should be added, as you need to load AutoDoc for the AutoDoc  command.

File: makedoc.g (hunk 9 above)
Comment (sebasguts) [reply makedoc.g]: Also, why do we need the `xy` package for AutoDoc? Maybe you accidentally copied the kbmag makedoc into this PR?

**Comment by sebasguts** [comment]
Also, some comments from me. Thanks a lot. This would then add to https://github.com/gap-packages/AutoDoc/issues/101

**Comment by cdwensley** [comment]
Just found the answer to my question in Magic.gd:  bib := "bib.xml",

File: PackageInfo.g (hunk 2 above)
Comment (cdwensley) [reply PackageInfo.g]: 2017 done.  Leaving """ to you!

File: PackageInfo.g (hunk 1 above)
Comment (cdwensley) [reply PackageInfo.g]: Now omitted - this was added before the checklist existed.

File: doc/Tutorials.xml (hunk 6 above)
Comment (cdwensley) [reply doc/Tutorials.xml]: Extra comment not yet added - this sentence is about transferring existing entities.

File: doc/Tutorials.xml (hunk 10 above)
Comment (cdwensley) [reply doc/Tutorials.xml]: Both now loaded

**Comment by cdwensley** [comment]
@fingolfin it was while I was adding some Index entries that I discovered that there were a large number of Index entries such as "@Description" which did not appear in the Index in manual.pdf.  So I reported this to Frank who has make a fix in GAPDoc-1.6.1.dev.  Now, when I run makedoc.g, I see these missing entries, but I also get a load of non-resolved references, such as 
     #W WARNING: non resolved reference: rec(
       BookName := "gapdoc",
       Func := "MakeGAPDocDoc" )
and I have no idea what to do about them. 
This problem is really not relevant to this PR - sorry for confusing things!

**Comment by fingolfin** [comment]
@cdwensley These are references into the GAPDoc manual. I do not get such warnings, but perhaps you installed a newer GAPDoc version from git but did build its manual? That might cause GAPDoc to fail to resolve the cross references.

In any case I do not get these warnings when building the AutoDoc manual in this branch of yours.

File: doc/Tutorials.xml [hunk 11]
```diff
@@ -313,6 +313,117 @@ AutoDoc( rec( gapdoc := rec( gap_root_relative_path := true ) ) );
...
+<Item> 
+Now edit <File>PackageName/PackageInfo.g</File> as follows. 
+  <List>
+  <Item> 
+  Delete any <Code>PKGVERSIONDATA</Code> chunk that may be there. 
+  The main reason for converting your manual to use &AutoDoc; is to stop 
+  using entities such as <Code>PACKAGENAMEVERSION</Code>. 
+  </Item>
+  <Item>
+  Add <Code>[ "AutoDoc", ">= &VERSION;" ]</Code>
```
Comment (fingolfin) [inline doc/Tutorials.xml]: Do we really want to suggest that minimal version?

Or, for that matter, do we really want people to add AutoDoc to their `SuggestedOtherPackages`? I see no advantage in doing that.

File: makedoc.g [hunk 12]
```diff
@@ -15,7 +15,18 @@ LoadPackage("AutoDoc");
...
-        includes := [ "Tutorials.xml", "Comments.xml" ]
+        includes := [ "Tutorials.xml", 
+                      "Comments.xml" ],
+        bib := "bib.xml", 
+        gapdoc_latex_options := rec( EarlyExtraPreamble := """
+            \usepackage[all]{xy} 
+            \newcommand{\bbZ} {\mathbb{Z}}
+        """ ),  
+        entities := rec( 
+            VERSION := "2017.09.15",
```
Comment (fingolfin) [inline makedoc.g]: Huh? Is this supposed to be the current version? Then I strongly object, as we are *guaranteed* to forget to update it with a release (though I guess that could be fixed by changing it to `VERSION := ~.Version`).

Or is it supposed to be "some version number, possibly old, which people should set as minimal required AutoDoc version? In that case, the name `VERSION` isn't so great, and there should also be a comment here explaining that use.

File: makedoc.g [hunk 13]
```diff
@@ -15,7 +15,17 @@ LoadPackage("AutoDoc");
 AutoDoc(rec( 
     autodoc := true,
     scaffold := rec(
-        includes := [ "Tutorials.xml", "Comments.xml" ]
+        includes := [ "Tutorials.xml", 
+                      "Comments.xml" ],
+        bib := "bib.xml", 
+        gapdoc_latex_options := rec( EarlyExtraPreamble := """
+            \usepackage[all]{xy}
```
[the remark this answers is not in the document]
Comment (cdwensley) [reply makedoc.g]: Just needed some suitable example here.  Now a4wide - any objections to that?

File: doc/Tutorials.xml (hunk 11 above)
Comment (cdwensley) [reply doc/Tutorials.xml]: My mistake - removed the instruction to put AutoDoc in SuggestedOtherPackasges . . .

File: makedoc.g (hunk 12 above)
Comment (cdwensley) [reply makedoc.g]: . . . and therefore removed entity VERSION.

**Comment by fingolfin** [comment]
Thank you so much!

**Comment by sebasguts** [comment]
Thank you very much also from me :)

**Pull request merged**